### ISSUE TITLE
slideRemove having issues when removing specific slides

### ISSUE THREAD

**Issue by kuchkuch2000** [kenwheeler/slick#3737]
> short description of the bug / issue, provide more detail below.

I have a small array which contains a boolean for displaying a slide. When I set one of the slide variables to false, it works fine, but if I set more than one to false, slideRemove function either removes the wrong slide or just doesn't remove any at all.
====================================================================

###### `http://jsfiddle.net/2bsu6crt/1/`  
(JSFIDDLE Forked from template below)

use this jsfiddle to reproduce your bug: 
http://jsfiddle.net/simeydotme/fmo50w7n/ 
we will likely close your issue without it.

====================================================================

#### Steps to reproduce the problem

1. First set all but 1 slide state to true
2. You will see that it works fine
3. Then try changing different slide states to false and change slide_button_link to a false message.
4. You will see that those messages will still show.

====================================================================

#### What is the expected behaviour?

...  The false messages are not supposed to show.

====================================================================

#### What is observed behaviour?

... slickRemove removes the wrong slides 

====================================================================

#### More Details

- Which browsers/versions does it happen on?
Currently working with chrome but I believe occurs in all

- Which jQuery/Slick version are you using?
JQuery: https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js
Slick: //cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.min.js

- Did this work before?
Not unless I have only 1 false state set

**Comment by Snowbell92** [comment]
yep, happening for me too. I kinda hijacked someone else's jsfiddle (sorry! but my code is way to complicated!) https://jsfiddle.net/hLgbd40r/2/ but the behavior is visible here if `infinite` is set to `true`.
